### DISCUSSION TITLE
Option to scale HiDPI game window to match system scaling.

### DISCUSSION THREAD

**Discussion by oddforms-design** [godotengine/godot-proposals#9573] [Editor]
**The problem:** The game window does not match the scaling of a HiDPI operating system setting. Because of this the scale of the game window does not match the project window settings. I have to manually resize the window to match my display each time I run the debugger and manually scale up the canvas when creating a webGL export to force the correct size for hidpi users.

**Example:** In the godot UI, if I want a 960px wide game window, I set my project window to 960px. But since I am working on a HiDPI display that is simulating a lower resolution, the debug window appears 480px wide.

**Example 2:** I set the project to 960px and export to webGL. Users playing the game without a hidpi display see a 960px window. Users with a hi dpi display see a 480px game window. To workaround fix is to manually set the page CSS body to 960px and set the #canvas scale to 100%.

**Proposed Solution:**  An advanced setting to make the hidpi window match an operating system's simulated resolution instead of the true resolution.

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/issues/7968.
